lib/session.js

```diff
@@ -133,8 +133,8 @@ Outgoing.prototype.available = function () {
 };
 
 Outgoing.prototype.compute_max_payload = function (tag) {
-    if (this.connection.max_frame_size) {
-        return this.connection.max_frame_size - (50 + tag.length);
+    if (this.connection.options.max_frame_size) {
+        return this.connection.options.max_frame_size - (50 + tag.length);
```

Comment:
This is incorrect I believe. The value used is that sent by the peer, not the locally configured value (that is the value that the peer in turn has to honour when sending).

Reply:
Thanks. I did not look closely enough. The test case set the same value to both ends of the connection.

I now checked. It is fragmenting before it has received the open from the other end, so it does not know the max_frame_size. This needs some further thought, but really is not the point why I started to make these changes so hard to justify spending time.

But this is clearly wrong. I think it must at least change it to `this.connection.remote.open.max_frame_size` with checks to see if the value is available.

Reply:
The max_frame_size property is added to connection as essentially an alias for remote.open.max_frame_size: https://github.com/amqp/rhea/blob/main/lib/connection.js#L249

That said your point about sending before this is known is a valid one. That can probably be most easily fixed in the test (e.g. by waiting for credit before sending).

Reply:
The amqp 1.0 defines MIN-MAX-FRAME-SIZE 512. That should be used until the open frame has been received for sending the frames to get the connection open. If that is used also for fragmenting the transfer frame, the 1M test case will fail because it will try to send more than 2048 frames.

Reply:
I changed the test to wait for the sender to be open.

Reply:
Somehow your second comment did not reach me. Maybe no need to change the patch as it is now, as the alias makes it difficult to use the MIN-MAX-FRAME-SIZE before the max_frame_size is received.

Reply:
A transfer can't actually be sent until credit is received, which must come after the open has been received. The problem is that rhea as a library allows messages to be passed in for sending even if they cannot yet be sent over the wire at present, and in this case it then makes a decision about fragmentation that cannot be accurately made.

This can be 'fixed' in the application by waiting for the sender to open, for credit or even just for the connection to open. Ideally there would be something built in that did not require this. I don't think defaulting to the min-max-frame is the right approach though.

Reply:
Ok. Thanks. Good point.

I'll see if I find some time for this. The fragmentation for sending is done too early. For a large message I see no reason to build the fragments beforehand. I did the change to send fragments in `Outgoing.prototype.process`, but I believe it could be changed to create the fragments there as needed rather than create all fragments beforehand. There is a condition on the credit before it gets to create the fragment, so it would solve this at the same time. I'd like to see how that would work out even though it is not critical for the performance.